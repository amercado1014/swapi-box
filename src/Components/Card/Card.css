.card {
  width: 275px;
  min-width: 220px;
  color: #fff;
  background-color: #000;
  margin: 15px 15px;
  padding: 10px;
  text-align: center;
  font-family: 'Anton', sans-serif;
  font-size: 20px
}

.star-button {
  height: 40px;
  width: 40px;
  background-color:  #fff;
  margin: auto;
}

.active {
  background-color:  #feda4a;
}

.people-card {
  border: 10px solid #00ff7f;
}

.planet-card {
  border: 10px solid #87ceeb;
}

.vehicle-card {
  border: 10px solid #e35757;
}

