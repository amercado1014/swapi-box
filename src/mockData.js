const mockCardData = {
  name: "Luke Skywalker",
  class: "people-card",
  data: {
    homeworld: "Tatooine",
    language: "Galactic Basic",
    population: "200000",
    species: "Human"
  }
};

const mockCardContainerData = [
  {
    name: "Luke Skywalker",
    class: "people-card",
    data: {
      homeworld: "Tatooine",
      language: "Galactic Basic",
      population: "200000",
      species: "Human"
    }
  },
  {
    name: "Darth Vader",
    class: "people-card",
    data: {
      homeworld: "Tatooine",
      language: "Galactic Basic",
      population: "200000",
      species: "Human"
    }
  },
  {
    name: "C-3PO",
    class: "people-card",
    data: {
      homeworld: "Tatooine",
      language: "n/a",
      population: "200000",
      species: "Droid"
    }
  }
];

const mockSummaryData = {
  date: "1983-05-25",
  episode: 6,
  text: "Luke Skywalker has returned to\r\nhis home planet of Tatooine",
  title: "Return of the Jedi"
};

export {
  mockCardData,
  mockCardContainerData,
  mockSummaryData
};