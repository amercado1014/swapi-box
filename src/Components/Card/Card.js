import React from 'react';
import './Card.css';
import PropTypes from 'prop-types';
import Icon from 'react-icons-kit';
import { starFull } from 'react-icons-kit/icomoon';

const Card = ({card, setFavorites, active}) => {
  const keys = Object.keys(card.data);
  const cardInfo = keys.map((key, index) => {
    return <p key={index}>{key}: {card.data[key]}</p>;
  });
  return (
    <div className={`card ${card.class}`}>
      <button className={`star-button ${active}`}  
        onClick={() => setFavorites(card)}>
        <Icon size={20} icon={starFull} />
      </button>
      <h2>{card.name}</h2>
      {cardInfo}
    </div>
  );
};

Card.propTypes = {
  card: PropTypes.object,
  setFavorites: PropTypes.func,
  active: PropTypes.string
};

export default Card;