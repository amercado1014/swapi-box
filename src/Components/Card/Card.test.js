import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme';
import { mockCardData } from '../../mockData';

describe('Card', () => {
  let wrapper;
  let mockSetFavorites;

  beforeEach(() => {
    mockSetFavorites = jest.fn();
    wrapper = shallow(<Card card={mockCardData}
      setFavorites={mockSetFavorites} />);
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have the correct number of p tags for card data', () => {
    expect(wrapper.find('p').length).toEqual(4);
  });

  it('should invoke setFavorites with the correct params', () => {
    const favoritesButton = wrapper.find('button');
    favoritesButton.simulate('click');
    expect(mockSetFavorites).toHaveBeenCalledWith(mockCardData);
  });
});