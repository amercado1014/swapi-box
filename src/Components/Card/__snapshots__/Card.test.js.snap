// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    card={
        Object {
            "class": "people-card",
            "data": Object {
              "homeworld": "Tatooine",
              "language": "Galactic Basic",
              "population": "200000",
              "species": "Human",
            },
            "name": "Luke Skywalker",
          }
    }
    setFavorites={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="star-button undefined"
          onClick={[Function]}
>
          <Icon
                    fill="currentColor"
                    icon={
                              Object {
                                        "children": Array [
                                          Object {
                                            "attribs": Object {
                                              "d": "M16 6.204l-5.528-0.803-2.472-5.009-2.472 5.009-5.528 0.803 4 3.899-0.944 5.505 4.944-2.599 4.944 2.599-0.944-5.505 4-3.899z",
                                              "fill": "#000000",
                                            },
                                            "name": "path",
                                          },
                                        ],
                                        "viewBox": "0 0 16 16",
                                      }
                    }
                    size={20}
          />
</button>,
        <h2>
          Luke Skywalker
</h2>,
        Array [
          <p>
            homeworld
            : 
            Tatooine
</p>,
          <p>
            language
            : 
            Galactic Basic
</p>,
          <p>
            population
            : 
            200000
</p>,
          <p>
            species
            : 
            Human
</p>,
        ],
      ],
      "className": "card people-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Icon
            fill="currentColor"
            icon={
                        Object {
                                    "children": Array [
                                      Object {
                                        "attribs": Object {
                                          "d": "M16 6.204l-5.528-0.803-2.472-5.009-2.472 5.009-5.528 0.803 4 3.899-0.944 5.505 4.944-2.599 4.944 2.599-0.944-5.505 4-3.899z",
                                          "fill": "#000000",
                                        },
                                        "name": "path",
                                      },
                                    ],
                                    "viewBox": "0 0 16 16",
                                  }
            }
            size={20}
/>,
          "className": "star-button undefined",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "fill": "currentColor",
            "icon": Object {
              "children": Array [
                Object {
                  "attribs": Object {
                    "d": "M16 6.204l-5.528-0.803-2.472-5.009-2.472 5.009-5.528 0.803 4 3.899-0.944 5.505 4.944-2.599 4.944 2.599-0.944-5.505 4-3.899z",
                    "fill": "#000000",
                  },
                  "name": "path",
                },
              ],
              "viewBox": "0 0 16 16",
            },
            "size": 20,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Luke Skywalker",
        },
        "ref": null,
        "rendered": "Luke Skywalker",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "homeworld",
            ": ",
            "Tatooine",
          ],
        },
        "ref": null,
        "rendered": Array [
          "homeworld",
          ": ",
          "Tatooine",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "language",
            ": ",
            "Galactic Basic",
          ],
        },
        "ref": null,
        "rendered": Array [
          "language",
          ": ",
          "Galactic Basic",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "population",
            ": ",
            "200000",
          ],
        },
        "ref": null,
        "rendered": Array [
          "population",
          ": ",
          "200000",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "species",
            ": ",
            "Human",
          ],
        },
        "ref": null,
        "rendered": Array [
          "species",
          ": ",
          "Human",
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="star-button undefined"
            onClick={[Function]}
>
            <Icon
                        fill="currentColor"
                        icon={
                                    Object {
                                                "children": Array [
                                                  Object {
                                                    "attribs": Object {
                                                      "d": "M16 6.204l-5.528-0.803-2.472-5.009-2.472 5.009-5.528 0.803 4 3.899-0.944 5.505 4.944-2.599 4.944 2.599-0.944-5.505 4-3.899z",
                                                      "fill": "#000000",
                                                    },
                                                    "name": "path",
                                                  },
                                                ],
                                                "viewBox": "0 0 16 16",
                                              }
                        }
                        size={20}
            />
</button>,
          <h2>
            Luke Skywalker
</h2>,
          Array [
            <p>
              homeworld
              : 
              Tatooine
</p>,
            <p>
              language
              : 
              Galactic Basic
</p>,
            <p>
              population
              : 
              200000
</p>,
            <p>
              species
              : 
              Human
</p>,
          ],
        ],
        "className": "card people-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Icon
              fill="currentColor"
              icon={
                            Object {
                                          "children": Array [
                                            Object {
                                              "attribs": Object {
                                                "d": "M16 6.204l-5.528-0.803-2.472-5.009-2.472 5.009-5.528 0.803 4 3.899-0.944 5.505 4.944-2.599 4.944 2.599-0.944-5.505 4-3.899z",
                                                "fill": "#000000",
                                              },
                                              "name": "path",
                                            },
                                          ],
                                          "viewBox": "0 0 16 16",
                                        }
              }
              size={20}
/>,
            "className": "star-button undefined",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "fill": "currentColor",
              "icon": Object {
                "children": Array [
                  Object {
                    "attribs": Object {
                      "d": "M16 6.204l-5.528-0.803-2.472-5.009-2.472 5.009-5.528 0.803 4 3.899-0.944 5.505 4.944-2.599 4.944 2.599-0.944-5.505 4-3.899z",
                      "fill": "#000000",
                    },
                    "name": "path",
                  },
                ],
                "viewBox": "0 0 16 16",
              },
              "size": 20,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Luke Skywalker",
          },
          "ref": null,
          "rendered": "Luke Skywalker",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "homeworld",
              ": ",
              "Tatooine",
            ],
          },
          "ref": null,
          "rendered": Array [
            "homeworld",
            ": ",
            "Tatooine",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "language",
              ": ",
              "Galactic Basic",
            ],
          },
          "ref": null,
          "rendered": Array [
            "language",
            ": ",
            "Galactic Basic",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "population",
              ": ",
              "200000",
            ],
          },
          "ref": null,
          "rendered": Array [
            "population",
            ": ",
            "200000",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "species",
              ": ",
              "Human",
            ],
          },
          "ref": null,
          "rendered": Array [
            "species",
            ": ",
            "Human",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
